README.md

```diff
@@ -1 +1,174 @@
-# arxiv-compiler
\ No newline at end of file
+# arXiv compiler service
+
+The build service compiles LaTeX source packages into PDF, PostScript,
+and other formats. This service encompasses the arXiv TeX tree. Compilation
+logs are also made available, for example to provide submitters feedback about
+compilation failures or warnings.
+
+There are three moving parts:
+- The compiler service API, a Flask web app that handles new requests for  
+  compilation and makes the results available (or not).
+- The compiler worker, a Celery app that grabs sources, dispatches compilation,
+  and stores the results.
+- The converter Docker image, which does the actual work of compilation and is
+  executed by the compiler worker. This contains the arXiv TeX tree.
+
+In addition, the following infrastructure parts are required:
+- Redis, used as a task queue between the API and the worker.
+- S3, used to store the result of compilation tasks.
+
+## TODO
+
+- [ ] Update the ``schema/``, and implement resource URLs in the response data.
+
+## Running the compiler service locally
+
+The easiest way to get up and running is to launch the whole service group
+using Docker Compose. You will need to pull the converter image ahead of time.
+
+If you do not have an instance of the file manager service running, you can
+try compiling published sources on the public arXiv.org site.
+
+You will also need a directory that the worker can use as /tmp space.
+
+For example:
+
+```bash
+$ mkdir /tmp/compilestuff     # Docker needs access to this.
+$ export HOST_SOURCE_ROOT=/tmp/compilestuff
+$ export CONVERTER_DOCKER_IMAGE=[name (including transport) of converter image]
+$ export FILEMANAGER_ENDPOINT=https://arxiv.org  # Get public sources.
+$ export FILEMANAGER_CONTENT_PATH=/src/{source_id}
+```
+
+And then run with:
+
+```bash
+$ docker-compose build    # Build the local images.
+$ docker-compose up       # Start the service group.
+```
+
+Give it a few seconds; Localstack needs to come up (provides a local S3), and
+a bucket will be created.
+
+### Authentication + Authorization
+
+To use the API you will need an auth token with scopes ``compile:read`` and
+``compile:create``. The easiest way to generate one of these is to use the
+helper script
+[here](https://github.com/arXiv/arxiv-auth/blob/develop/generate_token.py).
+Make sure that you use the same ``JWT_SECRET`` that is used in
+``docker-compose.yml``.
+
+
+```bash
+JWT_SECRET=foosecret pipenv run python generate_token.py
+```
+
+You should pass this token as the value of the ``Authorization`` header in
+all requests to the API. For example:
+
+```bash
+curl -XPOST -H "Authorization: [auth token]" http://127.0.0.1:8000/...
+```
+
+For requests in your browser, you can use something like
+[requestly](https://chrome.google.com/webstore/detail/requestly-redirect-url-mo/mdnleldcmiljblolnjhpnblkcekpdkpa?hl=en)
+to automatically add the auth header to your requests.
```

Comment:
I've also found [Advanced REST Client](https://install.advancedrestclient.com/install) to be handy for this purpose.

Reply:
Definitely! added in 3433449